src/dom.ts

```diff
@@ -35,7 +35,14 @@ export function fromTreeProto(ast: TreeProto) {
     },
   });
 
-  fromTreeProtoHelper(ast.tree, doc, doc);
+  if (ast.tree[0]?.tagid !== getTagId('ZERO_LENGTH')) {
+    const firstNode = ast.tree?.[0]?.tagid;
+    throw new Error(
+      `HTML must begin with a #document tag, found: ${firstNode}`
+    );
+  }
```

Comment:
```suggestion
  const firstNodeTagId = const ast.tree?.[0]?.tagid;
  if (firstNodeTagId !== getTagId('ZERO_LENGTH')) {
    throw new Error(
      `HTML must begin with a #document tag, found: ${firstNodeTagId}`
    );
  }
```

Reply:
Is writing `const` on the RHS of an assignment valid c++ 😛 ?
I like this pattern, Ill update to it...but without the const

Reply:
That was a typo lol